src/lib/comms/include/sol-socket.h

```diff
@@ -0,0 +1,315 @@
+     *     @c false otherwise.
+     */
+    int (*set_read_monitor)(struct sol_socket *s, bool on);
+
+    /**
+     * @brief Starts or stops monitoring the socket for writing.
+     *
+     * @li @c s the socket pointer (this)
+     * @li @c on @c true to be called when there is possible to write or
+     *     @c false otherwise.
+     */
+    int (*set_write_monitor)(struct sol_socket *s, bool on);
+
+    /**
+     * @brief Function to be called to read data from socket.
+     *
+     * @li @c s the socket pointer (this)
+     * @li @c buf the data buffer that will be used to receive the data
+     * @li @c len the size of @a buf
+     * @li @c cliaddr the source address of the message
+     */
+    ssize_t (*recvmsg)(struct sol_socket *s, void *buf, size_t len,
+        struct sol_network_link_addr *cliaddr);
+
+    /**
+     * @brief Function to be called to write data in the socket.
+     *
+     * @li @c s the socket pointer (this)
+     * @param buf the data to be transmitted
+     * @param len the size of @a buf
+     * @param cliaddr the address which the data will be sent
+     */
+    int (*sendmsg)(struct sol_socket *s, const void *buf, size_t len,
+        const struct sol_network_link_addr *cliaddr);
+
+    /**
+     * @brief Function to be called to join a multicast group.
+     *
+     * @li @c s the socket pointer (this)
+     * @li @c ifindex the index of the interface to be used.
+     * @li @c group the address of the group to join.
+     */
+    int (*join_group)(struct sol_socket *s, int ifindex,
+        const struct sol_network_link_addr *group);
+
+    /**
+     * @brief Function to be called to bind the socket to a network address.
+     *
+     * @li @c s the socket pointer (this)
+     * @li @c addr the address to bind.
+     */
+    int (*bind)(struct sol_socket *s, const struct sol_network_link_addr *addr);
+};
+
+/**
+ * @brief Structure to represent a socket.
+ *
+ * @see sol_socket_type
+ */
+struct sol_socket {
+#ifndef SOL_NO_API_VERSION
+#define SOL_SOCKET_API_VERSION (1)  /**< compile time API version to be checked during runtime */
+    /**
+     * must match #SOL_SOCKET_API_VERSION at runtime.
+     */
+    uint16_t api_version;
+#endif
+    const struct sol_socket_type *type;
+};
+
+/**
+ * @brief Creates an endpoint for communication.
+ *
+ * This function creates a socket using the system default implementation.
+ *
+ * @param options The socket's options..
+ *
+ * @return a handle to the socket on success, otherwise @c NULL is returned.
+ *
+ * @see sol_socket_del()
+ * @see sol_socket_default_options
+ */
+struct sol_socket *sol_socket_new(const struct sol_socket_default_options *options);
+
+/**
+ * @brief Destroy the @a socket instance.
+ *
+ * Destroy and release all the socket resources
+ *
+ * @param s The value got with @c sol_socket_new()
+ *
+ * @see sol_socket_new()
+ */
+void sol_socket_del(struct sol_socket *s);
+
+/**
+ * @brief Adds a function to be called when the socket had data
```

Comment:
oops, needs to fix the docs. Please refer to function specified on `new()` with config